mod-source-record-storage-server/src/main/java/org/folio/services/handlers/links/UpdateLinkProcessor.java

```diff
@@ -21,9 +21,14 @@ public class UpdateLinkProcessor implements LinkProcessor {
   @Override
   public Collection<Subfield> process(String fieldCode, List<org.folio.rest.jaxrs.model.Subfield> subfieldsChanges,
                                       List<Subfield> oldSubfields) {
+    if (isOnlyNaturalIdChanged(subfieldsChanges)) {
+      return processOnlyNaturalIdChange(oldSubfields, subfieldsChanges.get(0).getValue());
+    }
+
     var authorityIdSubfield = getAuthorityIdSubfield(oldSubfields);
     var authorityNaturalIdSubfield = getAuthorityNaturalIdSubfield(oldSubfields);
-    var controllableSubfields = new ArrayList<Subfield>();
+    var controllableAlphaSubfields = new ArrayList<Subfield>();
+    var controllableDigitSubfields = new ArrayList<Subfield>();
```

Comment:
I thought it was agreed to have this separation only for uncontrollable sufields. But controllable subfields should have same order as authority record.